src/analysis.ts

```diff
@@ -62,39 +62,39 @@ export class Analysis {
           `Expected 1.x.x, got ${analyzedPackage.schema_version}`);
     }
   }
+}
 
-  serialize(packagePath?: string): AnalyzedPackage {
-    const packageGatherer = new PackageGatherer();
-    const elementsGatherer = new ElementGatherer();
-    new AnalysisWalker(this._descriptors).walk([
-      packageGatherer, elementsGatherer
-    ]);
+export function generateElementMetadata(
```

Comment:
I would move this to it's own module, it could even just be in `elements-metadata.ts`